### PR TITLE
Add integration tests showing that bai-indexed traversal by intervals on cram files works

### LINKED ISSUE #675: Write basic integration tests for CRAM support using PrintReads

### PR DESCRIPTION
**Pull request by droazen**
-Add tests with an input cram and an output sam, bam, or cram with various intervals
 provided to the -L argument, and check that we get the expected reads in the output
 for each set of intervals.

-Generalize CRAMReferenceIntegrationTest to CRAMSupportIntegrationTest, and centralize
 cram test data in the engine package.

Resolves #675

### CODE REVIEW

**Comment by droazen** [comment]
For @edwardkw to review, please -- this is your big opportunity to exact revenge for all of the long code reviews we've put you through this summer! :)

File: src/test/java/org/broadinstitute/hellbender/engine/CRAMReferenceIntegrationTest.java [hunk 1]
```diff
@@ -1,82 +0,0 @@
...
-import org.broadinstitute.hellbender.tools.PrintReads;
-import org.broadinstitute.hellbender.utils.read.SamAssertionUtils;
-import org.broadinstitute.hellbender.utils.test.BaseTest;
-import org.testng.Assert;
-import org.testng.annotations.DataProvider;
-import org.testng.annotations.Test;
-
-import java.io.File;
-
-public final class CRAMReferenceIntegrationTest extends CommandLineProgramTest{
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/engine/CRAMReferenceIntegrationTest.java]: Note that the tests in the class weren't deleted -- the class was just renamed to `CRAMSupportIntegrationTest`, but github is failing to detect the rename.

File: src/test/java/org/broadinstitute/hellbender/engine/CRAMSupportIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        runCommandLine(args);
+    }
+
+    @DataProvider(name="testingDataNoRefMultipleInputs")
+    public Object[][] testingDataNoRefMulti() {
+        return new String[][]{
+                {"cramtest.sam", "cramtest.cram", ".bam"}
+        };
+    }
+// TODO: re-enable this test once wrong ref situation is fixed #793
```
Comment (edwardkw) [inline src/test/java/org/broadinstitute/hellbender/engine/CRAMSupportIntegrationTest.java]: Remove the giant comment; no longer needed.

**Comment by edwardkw** [comment]
Other than the one tiny thing, well-crafted tests. LGTM

File: src/test/java/org/broadinstitute/hellbender/engine/CRAMSupportIntegrationTest.java (hunk 2 above)
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/engine/CRAMSupportIntegrationTest.java]: Done

**Pull request merged**